### PR TITLE
feat: using keymap

### PR DESCRIPTION
**Pull request by danilshvalov**
Hi! I have to use different keyboard layouts, so I need to use `keymap`.  As far as I understand, `keymap` isn't supported in `leap.nvim`, so I decided to implement it.

P. S. I want to say thank you, because `leap.nvim` is amazing. 

Example:

https://user-images.githubusercontent.com/57654917/175834694-77c56863-fce8-48a8-843c-25d053845255.mp4

### CODE REVIEW

**Comment by ggandor** [comment]
Great, thanks! The problem is that this doesn't work when multiple characters should be converted to one (e.g. the keymap has something like `[i î`)... that seems tricky to handle.

**Comment by ggandor** [comment]
Sneak's implementation probably has the solution: https://github.com/justinmk/vim-sneak/blob/94c2de47ab301d476a2baec9ffda07367046bec9/autoload/sneak/util.vim#L21-L47

**Comment by danilshvalov** [comment]
I think I fixed this problem.

Example:

https://user-images.githubusercontent.com/57654917/176068488-287b3329-07ea-443c-884a-32ecc2df6c4c.mov

**Comment by danilshvalov** [comment]
In any case, they can be considered as English characters. So, user can just switch the `keymap`.

File: fnl/leap/main.fnl [hunk 1]
```diff
@@ -690,7 +707,7 @@ B: Two labels occupy the same position (this can occur at EOL or window
       (when force-no-labels? (inactivate-labels targets))
       (set-beacons targets {: force-no-labels?})
       (with-highlight-chores (light-up-beacons targets (inc idx)))
-      (match (or (get-input) (exit))
+      (match (or (get-input-by-keymap) (exit))
```
[the remark this answers is not in the document]
Comment (danilshvalov) [reply fnl/leap/main.fnl:710]: If a user uses the default labels, he must switch keymap. I don't think anyone wants to use non-English labels.

[the remark this answers is not in the document]
Comment (ggandor) [reply fnl/leap/main.fnl:712]: Yes, and that's exactly why we should not want to translate the keys at this phase - am I missing something?

[the remark this answers is not in the document]
Comment (danilshvalov) [reply fnl/leap/main.fnl]: I'm sorry. You're right. I was a little confused.

**Review by ggandor — APPROVED** [review]
Looks good, thanks!

**Comment by ggandor** [comment]
We should probably echo the input to the command line when keymap is active, but that's for another time.

**Pull request merged**